From 764a59e1136faf6dbb2e0caebe7304d7cc06fc29 Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Tue, 1 Sep 2026 02:29:11 +0800
Subject: [PATCH 3/3] Pass *dmem_cgroup_init to dmem_cgroup_register_region()
 on 7.3+

Commit e11640b9cea4e changed the function arguments of
dmem_cgroup_register_region() to except a struct *dmem_cgroup_init
instead of a u64.

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 kernel-open/nvidia/os-interface.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/kernel-open/nvidia/os-interface.c b/kernel-open/nvidia/os-interface.c
index 2bf5a5df55ab..bad60d5b792d 100644
--- a/kernel-open/nvidia/os-interface.c
+++ b/kernel-open/nvidia/os-interface.c
@@ -52,6 +52,8 @@
 #include <linux/cgroup_dmem.h>
 #endif // NV_DMEM_CGROUP_PRESENT
 
+#include <linux/version.h>
+
 extern char *NVreg_TemporaryFilePath;
 
 #define MAX_ERROR_STRING 528
@@ -2817,7 +2819,15 @@ NvU32 NV_API_CALL os_cgroup_implementation(void)
 
 void* NV_API_CALL os_dmem_cgroup_register_region(NvU64 size, const char *name)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 3, 0)
+    struct dmem_cgroup_init init = {
+        .size = size
+    };
+
+    void *region = dmem_cgroup_register_region(&init, name);
+#else
     void *region = dmem_cgroup_register_region(size, name);
+#endif
     if (IS_ERR(region))
     {
         return NULL;
-- 
2.55.0

